### ISSUE TITLE
Nuxt 3 app navigateTo in Middleware not working properly

### ISSUE THREAD

**Issue by Lengo46** [nuxt/nuxt#19994]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v18.12.1`
- Nuxt Version:     `3.1.1`
- Nitro Version:    `2.2.3`
- Package Manager:  `npm@9.4.2`
- Builder:          `vite`
- User Config:      `pages`, `css`, `components`, `extends`, `dir`, `typescript`, `apollo`, `postcss`, `build`, `runtimeConfig`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### auth.global.ts
```
export default defineNuxtRouteMiddleware(async (to, from) => {
  if (to.path === "/login") {
    return await navigateTo("/api/open-id/sso")
  }
}
```

### Describe the bug

This redirects to the available backend endpoint does redirect correctly in the browser but it gives a 404. If you just press enter in the browser (url bar) it does go to the route, also directly entering the url works.

![ ](https://user-images.githubusercontent.com/9341635/228799325-3e870500-39cb-4318-9059-367a85cd9d59.png)

![Screenshot 2023-03-30 at 11 52 20](https://user-images.githubusercontent.com/9341635/228799504-98373af9-7a4b-4a9b-88e4-47ef7c1004a9.png)

**Comment by danielroe** [comment]
If you are redirecting to a path outside your Vue app, you need to pass `{ external: true }` as a second parameter to `navigateTo`.

**Comment by Lengo46** [comment]
@danielroe thanks, but I've tried that and that isn't working.

**Comment by danielroe** [comment]
The fix only landed in v3.3.2: https://github.com/nuxt/nuxt/pull/19829.
